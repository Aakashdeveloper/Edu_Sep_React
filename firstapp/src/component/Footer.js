import React from 'react';

const Footer = (props) => {
    return(
        <React.Fragment>
            <hr/>
            <center>
                <h4>&copy; DeveloperFunnel {props.year} {props.month} </h4>
            </center>
        </React.Fragment>
    )
}

export default Footer;