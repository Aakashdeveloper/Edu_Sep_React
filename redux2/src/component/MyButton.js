import React from 'react';

class MyButton extends React.Component {
    constructor(){
        super()

        this.state={
            name:''
        }
    }

    onButtonClick = () => {
        this.setState({name:'SomeThing'})
    }

    render(){
        return(
            <div>
                <button className="mybutton" onClick={this.onButtonClick}>
                    {this.state.name}
                </button>
            </div>
        )
    }
}

export default MyButton;
