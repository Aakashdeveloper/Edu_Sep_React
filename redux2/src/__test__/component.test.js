import React from 'react';
import Footer from '../component/Footer';
import {create} from 'react-test-renderer';
import MyButton from '../component/MyButton';

describe('Testing Components',()=>{
    test('Testing Footer',() => {
        let tree = create(<Footer/>)
        expect(tree.toJSON()).toMatchSnapshot()
    });
    test("Testing My Button",() => {
        const component = create(<MyButton/>);
        const instance = component.root;
        const button = instance.findByType("button");
        button.props.onClick();
        expect(button.props.children).tobe('SomeThing')
    })
})