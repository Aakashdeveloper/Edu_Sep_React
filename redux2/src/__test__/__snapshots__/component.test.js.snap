// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Components Testing Footer 1`] = `
<footer>
  <div>
    <hr />
    Â© Developer Funnels
  </div>
</footer>
`;
